### PR TITLE
tidb: enable to forbid to retry when commit, add isolation test.

### PR DESCRIPTION
**Pull request by wentaoxu**
1. enable to forbid to retry when commit.
2. add isolation test.
PTAL @zhangjinpeng1987 @disksing @AndreMouche @siddontang @tiancaiamao

### CODE REVIEW

File: isolation_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+	session1.MustExec("begin;")
+	session1.MustExec("update x set c = c+1 where id = 1;")
+	session2.MustExec("begin;")
+	session2.MustQuery("select c from x where id = 1;").Check(testkit.Rows("1"))
+	session1.MustExec("commit;")
+	session2.MustExec("commit;")
+}
+
+func (s *testIsolationSuite) TestP2NonRepeatableRead(c *C) {
```
Comment (tiancaiamao) [inline isolation_test.go]: Maybe I was wrong, but from my understanding, this is "Read Skew", not "NonRepeatableRead" ?

File: isolation_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	session2.MustQuery("select c from x where id = 1;").Check(testkit.Rows("1"))
+	session2.MustExec("update x set c = c+1 where id = 1;")
+	session2.MustExec("commit;")
+	session1.MustExec("update x set c = c+1 where id = 1;")
+	_, err := session1.Exec("commit;")
+	c.Assert(err, NotNil)
+}
+
+// cursor is not supported
+func (s *testIsolationSuite) TestP4CLostUpdate(c *C) {}
```
Comment (tiancaiamao) [inline isolation_test.go]: ??

File: isolation_test.go (hunk 1 above)
Comment (wentaoxu) [reply isolation_test.go]: these cases are similar and confused, sometimes the queries are same except the order. but I make sure that all cases come from the paper<A Critique of ANSI SQL Isolation Levels>.

File: isolation_test.go (hunk 2 above)
Comment (wentaoxu) [reply isolation_test.go]: P4C is used for cursor which tidb doesn't support

File: isolation_test.go [hunk 3]
```diff
@@ -0,0 +1,250 @@
+package tidb_test
```
[the remark this answers is not in the document]
Comment (wentaoxu) [reply isolation_test.go]: ok

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**